src/Array.jl

```diff
@@ -119,6 +120,27 @@ function Matrix{T}(g::Gate{<:Control}) where {T}
     return M
 end
 
+function Base.rand(ElType::Type, ::Type{SU{N}}, lanes::NTuple{M, Int}; seed::Union{Int, Nothing}=nothing) where {N, M}
+    if seed !== nothing
+        Random.seed!(seed)
+    end
+
+    # Generate random numbers here
+    random_values = rand(ElType, N, N)
+
+    # Use QR to make it unitary
+    q, _ = qr(random_values)
+
+    SU{N}(lanes...; values = Matrix{ElType}(q))
+end
+
+Base.rand(::Type{SU{N}}, lanes::NTuple{M, Int}; seed::Union{Int, Nothing}=nothing) where {N, M} = rand(ComplexF64, SU{N}, lanes; seed = seed)
+Base.rand(::Type{Gate{SU{N}}}, lanes::Integer...; seed::Union{Int, Nothing}=nothing) where {N} = rand(ComplexF64, SU{N}, lanes; seed = seed)
```

Comment:
Is this function really needed? Doesn't this work without this specialization?

```suggestion
Base.rand(::Type{Gate{SU{N}}}, lanes...; kwargs...) where {N} = rand(SU{N}, lanes; kwargs...)
```

Reply:
Now that we changed `eltype` this is not necessary anymore